### ISSUE TITLE
Incorrect gas price: Estimated gas price: 3.2 gwei

### ISSUE THREAD

**Issue by JMariadlcs** [foundry-rs/foundry#6503]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### Operating System

None

### Describe the bug

I am trying to deploy a simple contract on Arbitrum using foundry but gas price is incorrectly calculated.

```bash
Chain 42161

Estimated gas price: 3.2 gwei

Estimated total gas used for script: 21628032

Estimated amount required: 0.0692097024 ETH
```

**Comment by Evalir** [comment]
This is a duplicate of many gas estimation issues—We cannot accurately estimate gas for L2s at this point in time. Will close to avoid duplicates—sorry!

**Comment by klkvr** [comment]
Hey @Evalir I believe that the problem here is not L2 semantics but just incorrect priority fee estimation.

Gas estimation may be difficult for L2s with unusual semantics like Optimism. However, currently, gas is estimated incorrectly for all chains where gas prices are just low. For example, for script running on Arbitrum, gas prices will always be estimated as ~3.2 gwei. However, actual gas price required is usually around 0.1 gwei.

This happens because gas estimations are done by using default ethers-rs eip1159 estimator. https://github.com/foundry-rs/foundry/blob/d4e6b43ea694f31ef2915e75d81c3308479ef8fd/crates/common/src/provider.rs#L280-L290

The code for default estimator can be found here https://github.com/gakonst/ethers-rs/blob/f0e5b194f09c533feb10d1a686ddb9e5946ec107/ethers-core/src/utils/mod.rs#L500.

The problem lies in this snippet:
```rust
  let max_priority_fee_per_gas =
      if base_fee_per_gas < U256::from(EIP1559_FEE_ESTIMATION_PRIORITY_FEE_TRIGGER) {
          U256::from(EIP1559_FEE_ESTIMATION_DEFAULT_PRIORITY_FEE)
      } else {
          std::cmp::max(
              estimate_priority_fee(rewards),
              U256::from(EIP1559_FEE_ESTIMATION_DEFAULT_PRIORITY_FEE),
          )
      };
```

`EIP1559_FEE_ESTIMATION_PRIORITY_FEE_TRIGGER` is 100 gwei and for `EIP1559_FEE_ESTIMATION_DEFAULT_PRIORITY_FEE` is 3 gwei.

So, basically, if base fee per gas is less than 100 gwei, the priority fee will be set to 3 gwei. This may be fine for Ethereum, but it is incorrect for chains with base gas price always being below 1 gwei (or even below 0.0000001 gwei on Base).

I think that replacing default estimator with some custom more flexible solution can help here.
